### PR TITLE
Feature/ Private key field

### PR DESCRIPTION
**Pull request by irubido**
InputField component: 
    added optional "eye" & "onEyeClick" props

InputPrompt component: 
    added reset on OK an Cancel buttons after Prompt closes
    added 0.5 sec delay on OK button (to show green border when password is correct)

PrivateKeyField component: 
    InputField props + password
    onEyeClick() opens PasswordPrompt checks for password and if correct shows Private key

### CODE REVIEW

File: src/renderer/components/PrivateKeyField/PrivateKeyField.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                onCancel={(): void=>{setShowPrompt(false);}}
+            />
+            <InputForm
+                label={props.label}
+                valid={props.valid}
+                errorMessage={props.errorMessage}
+                inputValue={privateKey}
+                placeholder={props.placeholder}
+                // onChange={(e): void => {setPrivateKey(e.currentTarget.value);props.onChange}}
+                onSubmit={(e): void => {e.preventDefault();props.onSubmit;}}
```
Comment (morrigan) [inline src/renderer/components/PrivateKeyField/PrivateKeyField.tsx]: Is this required? Can there be a submit action?

File: src/renderer/components/Prompt/InputPrompt.tsx [hunk 2]
```diff
@@ -44,12 +45,19 @@ export const InputPrompt: React.FunctionComponent<IInputPromptProps> = (props: I
...
+                        <ButtonDestructive onClick={(): void=>{
+                            setInputData("");
+                            props.onCancel();
+                        }}>
                             Cancel
                         </ButtonDestructive>
                     </div>
                     <div className={"prompt-confirm-button"}>
-                        <ButtonPrimary onClick={(): void => onSubmitWrapper()}>
+                        <ButtonPrimary onClick={(): void => {
```
Comment (morrigan) [inline src/renderer/components/Prompt/InputPrompt.tsx]: Will be easier to read if you separate it into function.

File: src/renderer/components/Prompt/InputPrompt.tsx [hunk 3]
```diff
@@ -44,12 +45,19 @@ export const InputPrompt: React.FunctionComponent<IInputPromptProps> = (props: I
                 />
                 <div className={"button-control"}>
                     <div className={"prompt-cancel-button"}>
-                        <ButtonDestructive onClick={props.onCancel}>
+                        <ButtonDestructive onClick={(): void=>{
```
Comment (morrigan) [inline src/renderer/components/Prompt/InputPrompt.tsx]: Will be easier to read if you separate it into function.

File: src/renderer/components/PrivateKeyField/PrivateKeyField.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+import {ISubmitStatus} from "../Prompt/InputPrompt";
+
+export interface IPrivateKeyProps extends IInputFormProps {
+    password: string;
+}
+export const PrivateKeyField: React.FunctionComponent<IPrivateKeyProps> = (props: IPrivateKeyProps) => {
+    const[showPrompt,setShowPrompt]=useState<boolean>(false);
+    const[privateKey/*, setPrivateKey **/]=useState<string | undefined>(props.inputValue);
+    const[timeoutStatus, setTimeoutStatus]=useState<boolean>(false);
+    const[passwordType, setPasswordType]=useState<string>("password");
```
Comment (morrigan) [inline src/renderer/components/PrivateKeyField/PrivateKeyField.tsx]: if there can be only two types - password and text, is it better to use a boolean flag like `hidden` ?

File: src/renderer/components/PrivateKeyField/PrivateKeyField.tsx (hunk 1 above)
Comment (irubido) [reply src/renderer/components/PrivateKeyField/PrivateKeyField.tsx]: I'll remove both onSubmit() and onChange(), as component only shows private key

**Pull request merged**